test/evp_kdf_test.c

```diff
@@ -297,6 +297,95 @@ static int test_kdf_x963(void)
     return ret;
 }
 
+/* Two test vectors from RFC 8009 (AES Encryption with HMAC-SHA2 for Kerberos
+ * 5) appendix A. */
+static int test_kdf_kbkdf_8009_prf1(void)
+{
+    int ret = 1, i = 0;
+    EVP_KDF_CTX *kctx;
+    OSSL_PARAM params[5];
+    char *label = "prf", *digest = "sha256", *prf_input = "test";
+    unsigned char input_key[] = {
+        0x37, 0x05, 0xD9, 0x60, 0x80, 0xC1, 0x77, 0x28,
+        0xA0, 0xE8, 0x00, 0xEA, 0xB6, 0xE0, 0xD2, 0x3C,
+    };
+    unsigned char output[] = {
+        0x9D, 0x18, 0x86, 0x16, 0xF6, 0x38, 0x52, 0xFE,
+        0x86, 0x91, 0x5B, 0xB8, 0x40, 0xB4, 0xA8, 0x86,
+        0xFF, 0x3E, 0x6B, 0xB0, 0xF8, 0x19, 0xB4, 0x9B,
+        0x89, 0x33, 0x93, 0xD3, 0x93, 0x85, 0x42, 0x95,
+    };
+    unsigned char result[sizeof(output)] = { 0 };
+
+    params[i++] = OSSL_PARAM_construct_utf8_string(
+        OSSL_KDF_PARAM_DIGEST, digest, strlen(digest) + 1);
+    params[i++] = OSSL_PARAM_construct_octet_string(
+        OSSL_KDF_PARAM_KEY, input_key, sizeof(input_key));
+    params[i++] = OSSL_PARAM_construct_octet_string(
+        OSSL_KDF_PARAM_SALT, label, strlen(label));
+    params[i++] = OSSL_PARAM_construct_octet_string(
+        OSSL_KDF_PARAM_INFO, prf_input, strlen(prf_input));
+    params[i] = OSSL_PARAM_construct_end();
+
+    kctx = get_kdfbyname(SN_kbkdf);
+    ret &= TEST_ptr(kctx);
+    ret &= TEST_true(EVP_KDF_CTX_set_params(kctx, params));
+    ret &= TEST_int_gt(EVP_KDF_derive(kctx, result, sizeof(result)), 0);
+    ret &= TEST_mem_eq(result, sizeof(result), output, sizeof(output));
+
+    EVP_KDF_CTX_free(kctx);
+    return ret;
+}
+static int test_kdf_kbkdf_8009_prf2(void)
+{
+    int ret = 1, i = 0;
+    EVP_KDF_CTX *kctx;
+    OSSL_PARAM params[5];
+    char *label = "prf", *digest = "sha384", *prf_input = "test";
+    unsigned char input_key[] = {
```

Comment:
I'd make these `static const`, but it isn't very important.

Reply:
That would require a cast later on, so I'm disinclined to do that.